tests/SpecTests/Context.php

```diff
@@ -53,6 +53,30 @@ public static function fromCommandMonitoring(stdClass $test, $databaseName, $col
         return $o;
     }
 
+    public static function fromCrud(stdClass $test, $databaseName, $collectionName)
+    {
+        $o = new self($databaseName, $collectionName);
+
+        $clientOptions = isset($test->clientOptions) ? (array) $test->clientOptions : [];
+
+        if (isset($test->outcome->collection->name)) {
+            $o->outcomeCollectionName = $test->outcome->collection->name;
+        }
+
+        $o->defaultWriteOptions = [
+            'writeConcern' => new WriteConcern(WriteConcern::MAJORITY),
+        ];
```

Comment:
Was this (and `outcomeFindOptions`) actually needed? AFAIK, all of the CRUD tests read/write to the primary and never specified a custom read concern or read preference.

AFAIK, `defaultWriteOptions` was only used for transaction tests (per [implementation instructions](https://github.com/mongodb/specifications/tree/master/source/transactions/tests#use-as-integration-tests)).

Reply:
I added this after a test kept failing on a replicaset setup. I got suspicious after I couldn’t replicate the failures when step-debugging through the test and thought that the problem might be with the secondaries being read before they have the data.